### ISSUE TITLE
Allow react 0.14.0-beta1 as peer dependency

### ISSUE THREAD

**Issue by cesarandreu** [react-bootstrap/react-overlays#3]
Does anything break with react 0.14.0-beta1?

**Comment by taion** [comment]
Everything to do with context will break. We'd need to use `unstable_renderSubtreeIntoContainer` or whatever they're calling it this week.

**Comment by cesarandreu** [comment]
Oh, I wasn't aware. I don't expect I'll be using contexts with my modals or popovers, so it should be alright.

In the meantime I made a branch [react-0.14](https://github.com/cesarandreu/react-overlays/tree/react-0.14) which replaces React with ReactDOM. It removes incompatible modules so I'm fairly certain that docs won't build... But since it's just for myself, it's alright for now. At least all the tests pass. I had to update a test, because findRenderedDOMComponentWithTag only accepts composite component.

I made another branch in which I added the built files: [react-0.14-build1](https://github.com/cesarandreu/react-overlays/tree/react-0.14-build1). In case someone wanted to play with it. 

No clue how people will be able to make libs that work with both v0.13 and v0.14...
